### ISSUE TITLE
[textfield] a11y review

### ISSUE THREAD

**Issue by Westbrook** [adobe/spectrum-web-components#1281]
### Feedback

- aria-label is not the optimal labelling mechanism for form fields. Please show an example which labels using visible text.
- invalid textfield does not include either html invalid or aria-invalid on the input element.
- Need examples which include things like hint text.
- invalid is not read on the invalid example

### Path forward

1. Expand documentation with the use of `sp-field-label`, we should prioritize this in the documentation 
2. Add appropriate `invalid` attributes
3. I do not believe that hint text isn't currently supported by Spectrum CSS, but I'll confirm the timeline on that.
4. Should be addressed by the attributes above.

cc @jnurthen
